lib/subscription/batcher.ex

```diff
@@ -91,21 +126,46 @@ defmodule AshGraphql.Subscription.Batcher do
     end
   end
 
-  def handle_info({_task, {:sent, topic, _res}}, state) do
-    case state.batches[topic] do
-      %{timer: timer} when not is_nil(timer) ->
-        Process.cancel_timer(timer)
+  def handle_info({_task, {:sent, topic, notifications, _res}}, state) do
+    batch = state.batches[topic]
 
-      _ ->
-        :ok
-    end
+    # remove the notifications that where sent from the batch
+    # it is possible that new notifications where added while
+    # the tasks processed the batch, so we need to remove only
+    # the notifications that where sent
+    batch =
```

Comment:
This would be a very expensive comparison. I think we need to figure something else out here, like move the batch to its own key, like "sending batches" that are keyed by their task. Comparing the list of notifications means a cross-wise comparison of every single value inside of every nested map/list etc.

Reply:
In that case can we just remove the batch when we start the task? If new notifications would be added it would just create a new batch. 
Right now, we don't do anything with the task that is saved to the batch except check if there is one when we send_all_batches and do nothing in that case. But that would be the same behaviour as removing the batch and not saving the task.

Reply:
Yeah, thats a good idea. We just have to *not* decrement the count until the batch has finished.

Reply:
I'll do that tomorrow.